datafusion/sqllogictest/test_files/select.slt

```diff
@@ -1129,5 +1129,35 @@ FROM t AS A, (SELECT * FROM t WHERE x = 0) AS B;
 0 0
 0 0
 
+query TT
+explain select coalesce(1, y/x), coalesce(2, y/x) from t;
```

Comment:
because of the reason describe in #8927, use plan to test.

Reply:
Can we please add some comments here explaining the rationale and what the expected outputs are (so future readers know if changes are expected)

Something like  this perhaps:

```
# Expressions that short circuit should not be refactored out as that may cause side effects (divide by zero)
# at plan time that would not actually happen during execution
```


Also, can you please add the tests that now pass (e.g. `select coalesce(1, y/x), coalesce(2, y/x) from t;`) so if someone breaks this code by accident, those queries would start failing, which might be easier to quickly tell is incorrect